pkg/network/ebpf/c/netns.h

```diff
@@ -0,0 +1,51 @@
+#ifndef __NETNS_H
+#define __NETNS_H
+
+#if defined(COMPILE_CORE) || defined(COMPILE_RUNTIME)
+
+#include "bpf_core_read.h"
+#include "bpf_telemetry.h"
+
+#ifdef COMPILE_RUNTIME
+#include <net/net_namespace.h>
+#include <net/sock.h>
+#endif
+
+#ifdef COMPILE_CORE
+#define sk_net __sk_common.skc_net
+#define CONFIG_NET_NS
+#endif
+
+static __always_inline u32 get_netns_from_ct_net(struct net* ct_net) {
+    u32 net_ns_inum = 0;
+#ifdef CONFIG_NET_NS
+#if defined(_LINUX_NS_COMMON_H) || defined(COMPILE_CORE)
```

Comment:
I'm having trouble understanding this clause -- when would we have COMPILE_RUNTIME be set but not _LINUX_NS_COMMON_H ?

Reply:
On older kernels. This is how this code was written before, but this needs to change going forward. In particular, on CO-RE we will take this branch, but on older kernels that will not work (since the field could be `proc_inum`). We need to have the kernel version available to make this decision on CO-RE, but that is WIP right now. We basically assume the field is `ns.inum` on CO-RE for now.